### DISCUSSION TITLE
External Javascript (not file include)

### DISCUSSION THREAD

**Discussion by dm-siegel** [GrapesJS/grapesjs#5099] [Q&A]
I have tried creating script block using the built in component (doesn't allow drop), but what I am really after is being able to add dynamic javascript (stored in db, not in a file) to the editor.  Using the includeScript when initializing the editor does not seem to work (when viewing the code).  I understand the security reasons for not allowing script tags... but is there a way to do it? (I don't necessarily need to be able to edit the script in the editor... just include it as part of the document).  I have considered making a custom tag that will hold the script source and using the view to render it in the editor, but was hoping for a more direct approach.

**Comment by artf** [comment]
There is the [`config.parser.optionsHtml.allowScripts` option](https://github.com/GrapesJS/grapesjs/blob/cb04b97cf6c6cd76676eb3aa095d5de54ef7bd1d/src/parser/config/config.ts#L78) to enable from the parser module

**Reply by dm-siegel** [reply]
I was doing this.... and it still seems to wipe the script when loading or viewing the code.

```typescript
this._editor = grapesjs.init({
      container: '#gjs',
      fromElement: true,
      // Size of the editor
      height: '100%',
      width: 'auto',
      // Disable the storage manager for the moment
      storageManager: false,
      optsHtml: {
        allowScripts: true,
      },
``` 

do I need to set it like:
```typescript
this._editor.config.parser.optionsHtml.allowScripts = true;
``` 
one difference i noticed is in the first block, I am setting "optsHtml" and in the second, I am setting "optionsHtml"

**Reply by dm-siegel** [reply]
to follow on to this... I am doing something like this:
```typescript
this._editor.setComponents("<script>alert('hello');</script>");
``` 

When I do this, the alert is not fired and the script tag is not showing up in the html when I view the code.

**Comment by dm-siegel** [comment]
more follow up... i was able to get it to accept the script tag using
```typescript
this._editor = grapesjs.init({
 // ...
 parser: {
   optionsHtml: { allowScripts: true }
 }
})
``` 

might have been referencing some old code trying to go through optsHtml?
